cybot/bot.py

```diff
@@ -159,52 +159,61 @@ def chat_slots(self, msg, *args):
         if(args[0]):
             wallet = Wallet(msg.username)
             timer = Timer(msg.username, 'slots')
-            cost = 0
+            bet = 0
             if args[0][0].isdigit():
-                cost = -abs(int(args[0][0]))
+                bet = -abs(int(args[0][0]))
             else:
                 self.sendmsg('Please place a numeric bet.')
                 return
 
             chk = timer.check(self.timeout['slots'])
             if(not chk['ready']):
-                timetil = int(chk['timetil'] / 60)
-                self.sendmsg('Try again in {} minute(s).'.format(timetil))
+                timetil = int(chk['timetil'])+1
+                self.sendmsg('Try again in {} second(s).'.format(timetil))
                 return
             else:
-                cost = abs(int(args[0][0]))
-                if wallet.balance >= cost:
-                    wallet.transaction(-cost)
+                bet = abs(int(args[0][0]))
+                if wallet.balance >= bet:
+                    wallet.transaction(-bet)
                     serverWallet = Wallet('{{server}}')
                     x = int(random.triangular(0, 6, 2))
                     z = int(random.triangular(0, 6, 2))
                     y = int(random.triangular(0, 6, 2))
                     prizemsg = ":botchat3:"
-                    translate = ['🍇', '🍒', '🍋', '🍌', '🂻', '♦']
+                    translate = ['🍇', '🍒', '🍋', '🍌', '💰', '💩']
                     prizemsg = "| {} | {} | {} |\n".format(
                         translate[x], translate[y], translate[z])
 
                     if 5 in (x, y, z) and (x == y == z):
-                        cost = serverWallet.balance
+                        prize = serverWallet.balance
                         serverWallet.transaction(-abs(serverWallet.balance))
+                        wallet.transaction(prize)
                         prizemsg += '{} hit the jackpot! They have earned {} squids!'.format(
                             msg.username, cost)
                     elif (x == y == z) and max(x, y, z) < 4:
-                        wallet.transaction(cost * 3)
-                        prizemsg += '{} matches 3 (three) fruits! [3x] Multiplyer (Bal: {})'.format(
-                            msg.username, wallet.balance)
+                        prize = (bet * 3)
+                        wallet.transaction(prize)
+                        serverWallet.transaction(-prize)
```

Comment:
If the server's wallet is responsible for the prizes there needs to be a rework of the probabilities. The suggestion could potentially cause a 0 jackpot eg. if someone were to place a bet 1/3 as large as the jackpot it's possible to drain the jackpot. (A drained jackpot isn't fun :/ )

Reply:
Most real slot machines have a max-bid limit to avoid that (since max payout is 3*bet, make max bet serverWallet/4?).   OTOH, real slot-machines are dealing with actual money, not squids, so it's more important to keep them fair.  

In light of the drained jackpot problem, might want to modify the prize for that to be 0.9 of serverWallet rather than the whole thing.